File: src/com/OpenSerenity/dsl/predicates/Be.java
package com.OpenSerenity.dsl.predicates;


import com.OpenSerenity.dsl.Assertion;
import com.OpenSerenity.elements.BaseElement;

public class Be {
    public static Assertion visible = new Assertion() {
        @Override
        public boolean invoke(BaseElement element) throws Exception {
            boolean result = element.isVisible();
            System.out.println(String.format("%s is visible %s", element.toString(), result));
            return result;
        }
    };
    public static Assertion invisible = new Assertion() {
        @Override
        public boolean invoke(BaseElement element) throws Exception {
            boolean result = element.isInVisible();
            System.out.println(String.format("%s is visible %s", element.toString(), result));
            return result;
        }
    };
}

File: src/scala/com/OpenSerenity/dsl/predicates/Be.scala
package com.OpenSerenity.dsl.predicates;

import com.OpenSerenity.elements.BaseElement

object Be {
  def visible() : BaseElement[_] => Boolean = (element : BaseElement[_]) => element.isVisible
  def inVisible() : BaseElement[_] => Boolean = (element : BaseElement[_]) => element.isInVisible
  def empty() : BaseElement[_] => Boolean = (element : BaseElement[_]) => element.isEmpty
  def notEmpty() : BaseElement[_] => Boolean = (element : BaseElement[_]) => !element.isEmpty
  def selected() : BaseElement[_] => Boolean = (element : BaseElement[_]) => element.isSelected
  def notSelected() : BaseElement[_] => Boolean = (element : BaseElement[_]) => !element.isSelected
}